### ISSUE TITLE
Codeceptjs 3.

### ISSUE THREAD

**Issue by cbruyndoncx** [thiagodp/concordialang-codeceptjs-core#5]
With the release of codecpetjs 3.0 there is a small syntax change
https://codecept.io/wiki/Upgrading-to-CodeceptJS-3.html

( I ) needs to become ( { I } )

I think the only required change is in the testgenerator file

**Comment by thiagodp** [comment]
Hello @cbruyndoncx , thank you for the suggestion. This will require upgrading `concordialang-codeceptjs-core` to version `2.x` and derived packages to use `codeceptjs@3`.

I'm also planning to release [Concordia 2](https://github.com/thiagodp/concordialang/issues/56) soon. Currently, version `1.x` works with `codeceptjs` 2.x and [*Concordia 2-alpha*](https://github.com/thiagodp/concordialang/issues/56).

**Comment by cbruyndoncx** [comment]
Yes, I actually started wit a codeceptjs3 installation and installed concordia alongside.
I am just starting with codeceptjs for a project and the combo with concordialang looks promising. Can't wait for v2.0
I was wondering though I found the BNF grammar, but is there an editor you can use that understands the BNF and the other concordialang syntax ?

**Comment by thiagodp** [comment]
Since Concordia is inspired by [Gherkin](https://specflow.org/bdd/gherkin/), you can use some useful [plugins](https://marketplace.visualstudio.com/items?itemName=stevejpurves.cucumber) for syntax highlighting.
Although, Concordia has its own reserved works (_e.g._, `Variant`, `Table`) and it would be great to have a fully-compatible plug-in. 
The [current documentation](https://github.com/thiagodp/concordialang/blob/master/docs/en/language.md) does not provide a BNF grammar. 

An old and incomplete draft goes like this:

```
feature ::= white comment tags header background? feature_elements comment?
header ::= (!(scenario_outline | scenario | background | variant_background) .)*
feature_elements ::= (scenario | scenario_outline)*
scenario ::= comment tags scenario_keyword space* lines_to_keyword white steps
scenario_outline ::= comment tags scenario_outline_keyword space* lines_to_keyword white steps testcase_sections white
background ::= comment background_keyword space* lines_to_keyword? (eol+ | eof) steps
variant_background ::= comment variant_background_keyword space* lines_to_keyword? (eol+ | eof) steps
tags ::= white (tag (space|eol)+)*
tag ::= '@' ([^@\r\n\t ])+
comment ::= (comment_line white)*
comment_line ::= space* '#' line_to_eol
steps ::= step*
step ::= comment step_keyword keyword_space line_to_eol (eol+ | eof) multiline_arg? white
testcase_sections ::= testcase*
testcase ::= comment space* testcase_keyword space* lines_to_keyword? eol table white
multiline_arg ::=  table | py_string
py_string ::= open_py_string (!close_py_string .)* close_py_string
open_py_string ::= space* '"""' space* eol
close_py_string ::= eol space* '"""' white
cell ::= [^\r\n|]+ '|'
row ::= space* '|' cell+ eol
table ::= row+
step_keyword ::= 'Given' | 'When' | 'Then' | 'And' | 'But'
import_keyword ::= 'Import'
testcase_keyword ::= 'Test Case:'
scenario_outline_keyword ::= 'Scenario Outline:'
scenario_keyword ::= 'Scenario:'
background_keyword ::= 'Background:'
variant_keyword ::= 'Variant:'
variant_background_keyword ::= 'Variant Background:'
property_line ::= '-' space* property connectors content
property ::= [^\r\n|]+
connectors ::= [^\r\n|]+
content ::= value | number
value ::= '"' [^\r\n|]+ '"'
number ::= [0-9]+(\.[0-9]*)?
table_keyword ::= 'Table:'
database_keyword ::= 'Database:'
constants_keyword ::= 'Constants:'
ui_element_keyword ::= 'UI Element:'
ui_element_step_keyword ::= 'Otherwise' | 'And' | 'But'
test_event ::= 'Before All' | 'After All' | 'Before Feature' | 'After Feature' | 'Before Each Scenario' | 'After Each Scenario'
lines_to_keyword ::= (!(eol space* reserved_words_and_symbols) .)*
reserved_words_and_symbols ::= (step_keyword keyword_space) | (variant_background_keyword keyword_space) | scenario_keyword | scenario_outline_keyword | variant_keyword | variant_background_keyword | table_keyword | database_keyword | constants_keyword | ui_element_keyword | test_event | table | tag | comment_line | property_line
line_to_eol ::= (!eol .)*
space ::= ' ' | '\t'
eol ::= '\r'? '\n'
white ::= (space | eol)*
keyword_space ::= ' '
```

Such BNF would cover the basic syntax. Concordia Compiler uses Natural Language Processing to analyze [actions](https://github.com/thiagodp/concordialang/blob/master/docs/en/actions.md) and other constructions.

**Comment by thiagodp** [comment]
Hi @cbruyndoncx , while the support for CodeceptJS 3 isn't available, you can upgrade the produced (CodeceptJS 2) tests automatically using [codecept3-upgrade](https://www.npmjs.com/package/codecept3-upgrade), like this:

```bash
npx codecept3-upgrade tests
```

**Comment by thiagodp** [comment]
@cbruyndoncx Released in v1.4
